### PR TITLE
PXD-1996 Feat/s3 listener

### PR DESCRIPTION
**Pull request by giangbui**
Description about what this pull request does.

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### New Features
Deploy ssjdispatcher

### CODE REVIEW

File: gen3/bin/kube-setup-roles.sh [hunk 1]
```diff
@@ -29,6 +29,13 @@ if [[ -z "$JENKINS_HOME" ]]; then
   if ! g3kubectl get rolebindings/devops-binding > /dev/null 2>&1; then
     g3kubectl apply -f "${GEN3_HOME}/kube/services/jenkins/rolebinding-devops.yaml"
   fi
+
+  if ! g3kubectl get serviceaccounts/jenkins-service > /dev/null 2>&1; then
```
Comment (frickjack) [inline gen3/bin/kube-setup-roles.sh]: why are you looking for the `jenkins-service` here?

File: kube/services/ssjdispatcher/rolebinding-devops.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+kind: RoleBinding
```
Comment (frickjack) [inline kube/services/ssjdispatcher/rolebinding-devops.yaml]: rename this file to `ssjdispatcher-binding.yaml`

File: gen3/bin/kube-setup-roles.sh [hunk 3]
```diff
@@ -29,6 +29,13 @@ if [[ -z "$JENKINS_HOME" ]]; then
...
+    g3kubectl apply -f "${GEN3_HOME}/kube/services/ssjdispatcher/serviceaccount.yaml"
+  fi
+  if ! g3kubectl get rolebindings/devops-binding > /dev/null 2>&1; then
```
Comment (frickjack) [inline gen3/bin/kube-setup-roles.sh]: you should be testing for the `ssjdispatcher-binding`, and rename the file

File: gen3/bin/kube-setup-secrets.sh [hunk 4]
```diff
@@ -102,7 +102,13 @@ if [[ (! -f "$configmapsFlagFile") || $(stat -c %Y "$configmapsFlagFile") -lt "$
   gen3 gitops configmaps
   touch "$configmapsFlagFile"
 fi
-
+# ssjdispatcher
+if ! g3kubectl get secrets/ssjdispatcher-creds > /dev/null 2>&1; then
```
Comment (frickjack) [inline gen3/bin/kube-setup-secrets.sh]: All of the other `-creds` secrets come from `creds.json` - just add an `ssjdispatcher` block to `creds.json` - do not introduce yet another secrets file

File: gen3/bin/kube-setup-roles.sh (hunk 1 above)
Comment (giangbui) [reply gen3/bin/kube-setup-roles.sh]: My bad copy! :(
Fixed in a55945ffa92f01cdd9dc040a8fa35656ee373220

File: kube/services/ssjdispatcher/rolebinding-devops.yaml (hunk 2 above)
Comment (giangbui) [reply kube/services/ssjdispatcher/rolebinding-devops.yaml]: Rename in a55945ffa92f01cdd9dc040a8fa35656ee373220

File: gen3/bin/kube-setup-roles.sh (hunk 3 above)
Comment (giangbui) [reply gen3/bin/kube-setup-roles.sh]: Tested and fixed in a55945ffa92f01cdd9dc040a8fa35656ee373220

File: gen3/bin/kube-setup-secrets.sh (hunk 4 above)
Comment (giangbui) [reply gen3/bin/kube-setup-secrets.sh]: Fixed in a55945ffa92f01cdd9dc040a8fa35656ee373220

**Pull request merged**